### ISSUE TITLE
请教一个Wta和Wtv计算问题

### ISSUE THREAD

**Issue by danache** [GeWu-Lab/OGM-GE_CVPR2022#2]
再demo code里，我看是这样子获取两个模态的输出的：
x, y, out = model(spec.unsqueeze(1).float(), image.float(), label, iteration)
out_v = (torch.mm(x,torch.transpose(model.module.fc_.weight[:,:512],0,1)) + model.module.fc_.bias/2)
out_a = (torch.mm(y,torch.transpose(model.module.fc_.weight[:,512:],0,1)) + model.module.fc_.bias/2)
loss = criterion(out, label)
我的问题是：
在这里显示的取前后512维度为两个模态的输出是否合理呢，因为out是是由一个FC得到的，而这个FC每一个输出都和所有输入相连的，人为的指定前512 channel是一个模态的参数我不太理解原因，恳请赐教。

**Comment by DavidInWuhanChina** [comment]
可能是因为融合前的维度是512吧？
[models](https://github.com/GeWu-Lab/OGM-GE_CVPR2022/tree/main/models)/fusion_modules.py中是这样写的：
class SumFusion(nn.Module):
    def __init__(self, **input_dim=512**, output_dim=100):
        super(SumFusion, self).__init__()
        self.fc_x = nn.Linear(input_dim, output_dim)
        self.fc_y = nn.Linear(input_dim, output_dim)

    def forward(self, x, y):
        output = self.fc_x(x) + self.fc_y(y)
        return x, y, output

**Comment by echo0409** [comment]
感谢 @DavidInWuhanChina 的回答！您的回答是正确的，因为encoder的输出是512维的向量，因此在拆分时按照前512和后512进行了拆分。此处具体数值可以根据您的encoder输出embedding的维度而定。

**Comment by pmj110119** [comment]
Thanks for your work! I have a question I hope you can answer.

I noticed that only one layer of FC was used in the code to fuse the two modalities, and this layer of FC was manually decoupled.

If there is a multi-layer FC, or a more complex head, can I still decompose the two modes in this way?
